pkg/pillar/cmd/zedmanager/zedmanager.go

```diff
@@ -467,8 +505,26 @@ func handleCreate(ctxArg interface{}, key string,
 			// FIXME - We really need a top level flag to tell the app is
 			//  a container. Deriving it from Storage seems hacky.
 			status.IsContainer = true
+			sha := utils.MaybeNameHasSha(sc.Name)
+			if sha != "" {
+				ss.ImageSha256 = sha
+				addAppAndImageHash(ctx, config.UUIDandVersion.UUID,
+					ss.ImageID, ss.ImageSha256)
+			} else {
+				log.Infof("Setting waitForResolve to true before publishing resolve config")
+				waitForResolve = true
+				status.RefCount++
```

Comment:
I think that field should be called HasResolverRef and not RefCount

Reply:
boolean is sufficient but it needs to be in StorageStatus and not in AppInstanceStatus

Reply:
Resolved.